File: my-folder/0735-asteroid-collision/solution.java
class Solution {
    public int[] asteroidCollision(int[] asteroids) {
        Stack<Integer> stack = new Stack<>();
        for (int i : asteroids) {
            if (i > 0) {
                stack.push(i);
            }
            while (!stack.isEmpty() && stack.peek() > 0 && stack.peek() < Math.abs(i)) {
                stack.pop();
            }
            if (stack.isEmpty() || stack.peek() < 0) {
                stack.push(i);
            }
            else if (i + stack.peek() == 0) {
                stack.pop();
            }
        }
        int[] res = new int[stack.size()];
        for (int i = res.length - 1; i >= 0; i--) {
            res[i] = stack.pop();
        }
        return res;
    }
}

File: my-folder/0735-asteroid-collision/solution.py
class Solution:
    def asteroidCollision(self, asteroids: List[int]) -> List[int]:
        stack = []

        for num in asteroids:
            if num < 0:
                while stack and stack[-1] > 0 and stack[-1] < abs(num):
                    stack.pop()
                if not stack or stack[-1] < 0:
                    stack.append(num)
                elif stack[-1] + num == 0:
                    stack.pop()
            else:
                stack.append(num)

        return stack
